include/genn/genn/snippet.h

```diff
@@ -119,11 +119,14 @@ class GENN_EXPORT Base
     //----------------------------------------------------------------------------
     // Structs
     //----------------------------------------------------------------------------
-    //! A variable has a name and a type
+    //! A variable has a name, a type and a readonly flag
+    /*! Through the wonders of C++ aggregate initialization, if readonly 
+    // is not initialized in the {}, it will be VALUE-initialised to false*/
```

Comment:
When I saw the syntax user-side, I was wondering whether a flag like "ro" vs "rw" in the initializer would have been more understandable ... but I can see how it looks very convenient from this side.
What do you think - any way to make this more explicit (i.e. that "true" in third position signifies readonly)?

Reply:
Yeah, in #55 the suggested syntax was adding "ro" or "rw" to the start of the type but, this is indeed much more convenient :) You're right that it's not very explicit though - maybe an enum might be better?

Reply:
An enum might work ...

Reply:
... and obviously I had forgotten #55 entirely.

Reply:
Hah me too

Reply:
The enum syntax is a little less terse but way clearer

Reply:
Ok, yes, this is much clearer. I think it's good to go.